shared/bar/src/commonMain/kotlin/com/tappchart/bar/BarChart.kt

```diff
@@ -1,27 +1,224 @@
     yAxisInterval: AxisInterval,
     data: List<Bar>,
-    verticalLabel: List<ValueLabel>,
+    verticalLabels: List<ValueLabel>,
     labelStyle: TextStyle,
     spacingBetweenBars: Dp,
     axisStyle: AxisStyle,
-    background: List<BackgroundStyle> = listOf(),
     barColor: List<Color>,
+    background: List<BackgroundStyle> = listOf(BackgroundStyle.HorizontalGrid),
+    yAxisPaddingToLine: Dp = 8.dp,
+    xAxisPaddingToBars: Dp = 2.dp,
+    firstBarStartPadding: Dp = 0.dp,
+    lastBarEndPadding: Dp = 0.dp,
+    showTopLabelMisaligned: Boolean = false
+) {
+
+    val textMeasurer = rememberTextMeasurer()
+    Canvas(modifier = modifier) {
+        val horizontalLabelHeight = textMeasurer.measure(data[0].key.label).size.height
+        val bottomOfChart = size.height - horizontalLabelHeight - xAxisPaddingToBars.toPx()
+        val maxWidthOfVerticalLabels =
+            verticalLabels.maxOf { textMeasurer.measure(it.label).size.width }.toFloat()
+
+        drawVerticalLabelsAndBackground(
+            textMeasurer = textMeasurer,
+            verticalLabels = verticalLabels,
+            labelStyle = labelStyle,
+            bottomOfChart = bottomOfChart,
+            background = background,
+            axisStyle = axisStyle,
+            yAxisInterval = yAxisInterval,
+            yAxisPaddingToLine = yAxisPaddingToLine.toPx(),
+            shouldShowTopLabelMisaligned = showTopLabelMisaligned,
+            maxWidthOfVerticalLabels = maxWidthOfVerticalLabels
+        )
+
+        drawBarsWithHorizontalLabels(
+            textMeasurer = textMeasurer,
+            bars = data,
+            barColor = barColor,
+            yAxisInterval = yAxisInterval,
+            bottomOfChart = bottomOfChart,
+            paddingBetweenBars = spacingBetweenBars.toPx(),
+            labelStyle = labelStyle,
+            xAxisPaddingToBars = xAxisPaddingToBars.toPx(),
+            firstBarStartPadding = firstBarStartPadding.toPx(),
+            lastBarEndPadding = lastBarEndPadding.toPx(),
+            yAxisPaddingToFirstBar = maxWidthOfVerticalLabels + yAxisPaddingToLine.toPx()
+        )
+    }
+}
+
+private fun DrawScope.drawBarsWithHorizontalLabels(
+    textMeasurer: TextMeasurer,
+    bars: List<Bar>,
+    barColor: List<Color>,
+    yAxisInterval: AxisInterval,
+    bottomOfChart: Float,
+    labelStyle: TextStyle,
+    paddingBetweenBars: Float,
+    yAxisPaddingToFirstBar: Float,
+    xAxisPaddingToBars: Float,
+    firstBarStartPadding: Float,
+    lastBarEndPadding: Float
 ) {
-}
\ No newline at end of file
+    val canvasWidthWithoutPadding =
+        size.width - yAxisPaddingToFirstBar - firstBarStartPadding - lastBarEndPadding
+    val barWidth = (canvasWidthWithoutPadding - paddingBetweenBars * bars.lastIndex) / bars.size
+
+    var barStartPosition = yAxisPaddingToFirstBar + firstBarStartPadding
+    bars.forEachIndexed { index, bar ->
+        val color = barColor.getOrNull(index) ?: barColor.first()
```

Comment:
Since barColor can be empty I think we sould use `.firstOrNull()` and also provide a default color 🤔